File: py/waktu.py
class Waktu :
    def __init__(self, jam, menit) :
        self.__jam=jam
        self.__menit=menit

    def setJam(self,jam) : self.__jam=jam
    def setMenit(self,menit) : self.__menit=menit

    def getJam(self) : return self.__jam
    def getMenit(self) : return self.__menit

    # def wrongInput(self) :
    #     return (
    #         (int(self.__jam) > 24 or int(self.__jam < 0))
    #         and
    #         (int(self.__menit) > 59 or int(self.__menit < 0))
    #     )

    def input(self) :
        self.__jam=input("Masukkan jam: ")
        self.__menit=input("Masukkan menit: ")
        # if(self.wrongInput()) :
        #     print("Waktu yang anda input tidak sesuai dengan ketentuan")
        #     return

    def toString(self) : return str(self.__jam) + ":" + str(self.__menit)

File: cpp/waktu.cpp
class Waktu {
    private:
        int jam;
        int menit;
        
    public:
        Waktu() {
            this->jam=0;
            this->menit=0;
        }

        Waktu(int jam, int menit, int detik) {
            this->jam=jam;
            this->menit=menit;
        }

        void setJam(int jam) { this->jam=jam; }
        void setMenit(int menit) { this->menit=menit; }

        int getJam() { return this->jam; }
        int getMenit() { return this->menit; }

        bool wrongInput() {
            return (
                (this->jam > 24 || this->jam < 0)
                &&
                (this->menit > 59 || this->menit < 0)
            );
        }

        void input() {
            cout << "Masukkan jam: "; cin >>this->jam;
            cout << "Masukkan menit: "; cin >> this->menit;
            if(this->wrongInput()) {
                cout << "Waktu yang anda input tidak sesuai dengan ketentuan";
                return;
            }
        }

        string toString() {
            return this->jam + ":" + this->menit;
        }
};